File: app/src/main/java/com/dev/bins/note/model/Note.java
package com.dev.bins.note.model;

import org.litepal.crud.DataSupport;

import java.util.Date;

/**
 * Created by bin on 11/24/15.
 */
public class Note extends DataSupport{

    private long id;
    private String title;
    private String content;
    private Date date;
    private boolean isShow;

    private Category category;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public boolean isShow() {
        return isShow;
    }

    public void setShow(boolean isShow) {
        this.isShow = isShow;
    }

    public Category getCategory() {
        return category;
    }

    public void setCategory(Category category) {
        this.category = category;
    }

    @Override
    public String toString() {
        return "Note{" +
                "id=" + id +
                ", title='" + title + '\'' +
                ", content='" + content + '\'' +
                ", date=" + date +
                ", isShow=" + isShow +
                ", category=" + category +
                '}';
    }
}

File: app/src/main/java/com/dev/bins/note/model/Note.kt
package com.dev.bins.note.model

import com.raizlabs.android.dbflow.structure.BaseModel
import java.util.*

/**
 * Created by bin on 11/24/15.
 */
class Note : BaseModel() {

    var id: Long = 0
    var title: String? = null
    var content: String? = null
    var date: Date? = null
    var isShow: Boolean = false

    var category: Category? = null

    override fun toString(): String {
        return "Note{" +
                "id=" + id +
                ", title='" + title + '\'' +
                ", content='" + content + '\'' +
                ", date=" + date +
                ", isShow=" + isShow +
                ", category=" + category +
                '}'
    }
}
